src/Adapter/PlatformServices.Desktop/Utilities/DeploymentUtility.cs

```diff
@@ -156,27 +144,37 @@ private bool Deploy(string source, IRunContext runContext, ITestExecutionRecorde
         /// <param name="deploymentDirectory">The deployment directory.</param>
-        /// <param name="deploymentWarnings">Warnings.</param>
+        /// <param name="resultsDirectory">Root results directory</param>
+        /// <returns>Returns a list of deployment warnings</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
-        private void Deploy(IList<DeploymentItem> deploymentItems, string testSource, string deploymentDirectory, out IEnumerable<string> deploymentWarnings)
+        private IEnumerable<string> Deploy(IList<DeploymentItem> deploymentItems, string testSource, string deploymentDirectory, string resultsDirectory)
         {
-            Debug.Assert(!string.IsNullOrEmpty(deploymentDirectory), "Deployment directory is null/empty");
-            Debug.Assert(this.fileUtility.DoesDirectoryExist(deploymentDirectory), "Deployment directory " + deploymentDirectory + " does not exist");
-            Debug.Assert(!string.IsNullOrEmpty(testSource), "TestSource directory is null/empty");
-            Debug.Assert(this.fileUtility.DoesFileExist(testSource), "TestSource " + testSource + " does not exist.");
+            Validate.IsFalse(string.IsNullOrWhiteSpace(deploymentDirectory), "Deployment directory is null or empty");
+            Validate.IsTrue(this.fileUtility.DoesDirectoryExist(deploymentDirectory), $"Deployment directory {deploymentDirectory} does not exist");
+            Validate.IsFalse(string.IsNullOrWhiteSpace(testSource), "TestSource directory is null/empty");
+            Validate.IsTrue(this.fileUtility.DoesFileExist(testSource), $"TestSource {testSource} does not exist.");
 
             testSource = Path.GetFullPath(testSource);
-
             var warnings = new List<string>();
 
-            // Get the referenced assemblies.
-            this.ProcessNewStorage(testSource, deploymentItems, warnings);
+            if (MSTestSettingsProvider.Settings.DeployTestSourceDependencies)
+            {
+                EqtTrace.Info("Adding the references and satellite assemblies to the deploymentitems list");
+
+                // Get the referenced assemblies.
+                this.ProcessNewStorage(testSource, deploymentItems, warnings);
 
-            // Get the satellite assemblies
-            var satelliteItems = this.GetSatellites(deploymentItems, testSource, warnings);
-            foreach (var satelliteItem in satelliteItems)
+                // Get the satellite assemblies
+                var satelliteItems = this.GetSatellites(deploymentItems, testSource, warnings);
+                foreach (var satelliteItem in satelliteItems)
+                {
+                    this.deploymentItemUtility.AddDeploymentItem(deploymentItems, satelliteItem);
+                }
+            }
+            else if (File.Exists(testSource))
```

Comment:
We already have this check in beginning of this Deploy() function. Probably we can skip it here.

Reply:
Done.